paasta_tools/contrib/service_shard_update.py

```diff
@@ -39,6 +39,13 @@ def parse_args():
         action="store_true",
         dest="verbose",
     )
+    parser.add_argument(
+        "-d",
+        "--dry-run",
+        help="Do not commit changes to git",
+        action="store_true",
+        dest="dry_run",
+    )
```

Comment:
++

(i'm also a fan of `--for-real` flags so that dry-run is the default, but that would be a little more annoying to rollout here)